src/lightning/pytorch/utilities/testing/_runif.py

```diff
@@ -177,6 +179,19 @@ def test_wrapper(arg1):
         conditions.append(_TORCH_GREATER_EQUAL_2_0 and not _ONNX_AVAILABLE)
         reasons.append("onnx")
 
+    if dynamo:
+        if _TORCH_GREATER_EQUAL_2_1:
+            from torch._dynamo.eval_frame import is_dynamo_supported
+
+            cond = not is_dynamo_supported()
+        else:
+            cond = sys.platform == "win32" or sys.version_info >= (3, 11)
+
+        conditions.append(cond)
+        # set use_base_version for nightly support
+        conditions.append(compare_version("torch", operator.lt, "2.0.0", use_base_version=True))
```

Comment:
There needs to be as many conditions as reasons, so you shouldn't append separately.

I suggest doing `cond &= compare_version(...)`

Reply:
doesn't it need to be logical or?